src/SALib/sample/morris/morris.py

```diff
@@ -132,19 +124,24 @@ def _sample_morris(problem: dict, number_trajectories: int,
     """
 
     group_membership, _ = compute_groups_matrix(problem['groups'])
-    check_group_membership(group_membership)
+    _check_group_membership(group_membership)
 
     num_params = group_membership.shape[0]
     num_groups = group_membership.shape[1]
 
-    sample = np.array([generate_trajectory(group_membership,
-                                           num_levels)
-                       for n in range(number_trajectories)])
-    return sample.reshape((number_trajectories * (num_groups + 1), num_params))
+    i = 0
+    sample_morris = []
+    while i < number_trajectories:
+        sample_morris.append(_generate_trajectory(group_membership, num_levels))
+        i += 1
+    sample_morris = np.array(sample_morris)
```

Comment:
While this is more readable than what it replaces, I wonder if it is necessary to introduce a counter.  `for n in range(number_trajectories)` seems sufficiently clear.  I would tidy this as follows:
```python
sample = [generate_trajectory(group_membership, num_levels) for n in range(number_trajectories)]
sample_morris = np.array(sample).reshape((number_trajectories * (num_groups + 1), num_params))
return sample_morris
```
But perhaps this is just bike-shedding...

Reply:
My "problem" with this approach is that `n` isn't actually used for anything, that's why I prefer the `while` over the `for` loop, even if I have to include a counter. Besides, I have the impression that using `reshape` makes the code more difficult to understand. It's up to you, though...

Reply:
Using `range()` is more pythonic in this case.

If the value (`n`) is not used then Python convention is to indicate that with an underscore (`_`), for example:

`[generate_trajectory(group_membership, num_levels) for _ in range(number_trajectories)]`

See [here](https://stackoverflow.com/a/47599668/2694952) for another use of the underscore.

Reply:
Great, @ConnectedSystems. I just made the change as you proposed.